lib/ssoe/service.rb

```diff
@@ -56,5 +55,61 @@ def build_message(credential_method, credential_id, user, address)
         zipcode: address.zipcode
       ).perform
     end
+
+    def parse_response(response_body)
+      parsed = Hash.from_xml(response_body)
+
+      if parsed.dig('Envelope', 'Body', 'getSSOeTraitsByCSPIDResponse', 'icn')
+        icn = parsed['Envelope']['Body']['getSSOeTraitsByCSPIDResponse']['icn']
+        return { success: true, icn: } if icn.present?
+      end
+
+      if parsed.dig('Envelope', 'Body', 'Fault')
+        fault_code = parsed.dig('Envelope', 'Body', 'Fault', 'faultcode')
+        fault_string = parsed.dig('Envelope', 'Body', 'Fault', 'faultstring')
+
+        return {
+          success: false,
+          error: {
+            code: fault_code || 'UnknownError',
```

Comment:
Same thing here, you can use `code: parsed.dig('Envelope', 'Body', 'Fault', 'faultcode') || 'UnknownError',`